assets/src/data/eventespresso/base-controls.js

```diff
@@ -63,6 +80,27 @@ const controls = {
 	DISPATCH( { reducerKey, dispatchName, args } ) {
 		return dispatchData( reducerKey )[ dispatchName ]( ...args );
 	},
+	RESOLVE_SELECT( { reducerKey, selectorName, args } ) {
+		return new Promise( ( resolve ) => {
+			const hasFinished = () => selectData( 'core/data' )
+				.hasFinishedResolution( reducerKey, selectorName, args );
+			const getResult = () => selectData( reducerKey )[ selectorName ]
+				.apply( null, args );
+
+			// trigger the selector (to trigger the resolver)
+			const result = getResult();
+			if ( hasFinished() ) {
+				return resolve( result );
```

Comment:
Because we do an initial trigger of the selector and only _use_ the value if it has resolved.

If I use `getResult()` directly, then I have to still call it outside of the `hasFinished()` conditional.  So this actually reduces the number of times the select has to be invoked.

I _might_ be able to get away with re-using the `result` value in the subscribe callback.

Reply:
Okay so the reason why things are the way they are is this:

- The initial `const result = getResult()` invokes the selector and if the selector has not been resolved, this triggers the resolution.  If it HAS resolved then it returns the resolved value which can then just be returned on line 93.  So `result` only gets returned if resolution is finished.  As mentioned earlier, assigning to the const allows for skipping another unnecessary call to the selector.
- the call to `resolve( getResult() )` in line 99 is inside the subscription callback.  We need to invoke the selector every time the subscription fires because that it will fire at various resolution states for that selector.  That's why we can't simply use the value on the `result` const because it will get stale.